library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,450 @@
+require "yast"
+
+require "abstract_method"
+
+module CWM
+  # Represent base for any widget used in CWM. It can be passed as "widget" argument. For more
+  # details about usage see {CWM.ShowAndRun}
+  class AbstractWidget
```

Comment:
Widget can get impression that it can be used itself as widget, but it is not true. instace of Widget itself is useless, so I add there abstract to make it explicit.